components/common/navigation/Context.tsx

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import { useEffect } from "react";
+import * as React from "react";
+
+export type NavigationParams = { [key: string]: string | undefined }
+export type Routes = Map<string, Array<undefined | ((params: NavigationParams) => void)>>;
+
+export interface NavigationContext {
```

Comment:
Especially because this is a shared module, this could benefit from some comments / documentation at least at the top